docs/reference/docs/delete-by-query.asciidoc

```diff
@@ -0,0 +1,308 @@
+[[docs-delete-by-query]]
+== Delete By Query API
+
+experimental[The delete-by-query API is new and should still be considered experimental.  The API may change in ways that are not backwards compatible]
+
+The simplest usage of `_delete_by_query` just performs a deletion on every
```

Comment:
I agree, this is a doc issue. The code checks for the presence of a query and do not allow `_delete_by_query` usage without a query.